nsxt/resource_nsxt_vpc_service_profile_test.go

```diff
@@ -0,0 +1,226 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+var accTestPolicyVpcServiceProfileCreateAttributes = map[string]string{
+	"display_name":     getAccTestResourceName(),
+	"description":      "terraform created",
+	"ntp_servers":      "5.5.5.5",
+	"lease_time":       "50840",
+	"mode":             "DHCP_IP_ALLOCATION_BY_PORT",
+	"dns_server_ips":   "7.7.7.7",
+	"server_addresses": "11.11.11.11",
+	"cache_size":       "2048",
+	"log_level":        "DEBUG",
+}
+
+var accTestPolicyVpcServiceProfileUpdateAttributes = map[string]string{
+	"display_name":     getAccTestResourceName(),
+	"description":      "terraform updated",
+	"ntp_servers":      "5.5.5.7",
+	"lease_time":       "148000",
+	"mode":             "DHCP_IP_ALLOCATION_BY_MAC",
+	"dns_server_ips":   "7.7.7.2",
+	"server_addresses": "11.11.11.111",
+	"cache_size":       "1024",
+	"log_level":        "INFO",
+}
+
+func TestAccResourceNsxtVpcServiceProfile_basic(t *testing.T) {
+	testResourceName := "nsxt_vpc_service_profile.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccOnlyMultitenancy(t) },
```

Comment:
TBH I'm not sure if that should be tested with multitenancy or with VPC env. For now definitely with VPC as it runs on v9.0 (these tests will fail with v4.1.2).

Reply:
I think this should be part of VPC test suite later on too